client/guest/guest-form.jsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+
+class GuestForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleClick = this.handleClick.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.getAllGuests = this.getAllGuests.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.state = {
+      isClicked: false,
+      firstName: '',
+      lastName: ''
+    };
+  }
+
+  handleClick() {
+    const isClicked = this.state.isClicked;
+    this.setState({
+      isClicked: !isClicked
+    });
+  }
+
+  handleChange(event) {
+    const target = event.target;
+    if (target.name === 'first-name') {
+      this.setState({
+        firstName: target.value
+      });
+    }
+    this.setState({
+      lastName: target.value
+    });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    const firstName = this.state.firstName;
+    const lastName = this.state.lastName;
+    const form = document.getElementById('guest-form');
```

Comment:
we don't query the DOM like this in React. is `event.target` what you're looking for?